Doc/c-api/exceptions.rst

```diff
@@ -461,10 +462,10 @@ Querying the error indicator
 
    .. deprecated:: 3.12
 
+      Use :c:func:`PyErr_GetRaisedException` instead.
 
-.. c:function:: void PyErr_Restore(PyObject *type, PyObject *value, PyObject *traceback)
 
-    As of 3.12, this function is deprecated. Use :c:func:`PyErr_SetRaisedException` instead.
+.. c:function:: void PyErr_Restore(PyObject *type, PyObject *value, PyObject *traceback)
 
    Set the error indicator from the three objects.  If the error indicator is
    already set, it is cleared first.  If the objects are ``NULL``, the error
```

Comment:
Maybe worth porting the heavily revised/clarified text you implemented for `PyErr_SetRaisedException` in PR #101962 to here as well, given this text is very similar to nearly identical (being the source of that one's)?

Reply:
PTAL!